### PR TITLE
feat(coins): add a settings to allow limitation of simultaneous coin enabled

### PR DESCRIPTION
**Pull request by Milerius**
first step for #884 

cc @SylEze  or @ssakone 

You can simply use the size of the portfolio model size to get the number of coins -> https://github.com/KomodoPlatform/atomicDEX-Desktop/blob/a80c2a3c7ea68e952bd97125987b5971753bd45b/src/core/atomicdex/models/qt.portfolio.model.hpp#L44

You can then use the new settings, and the number of checked coins in the enable modal to ensure that the user cannot enabled more than `MaximumNbCoinsEnabled`

You can make a new visual settings for changing this number with predefined values: [50, 75, 100, 150, 200]

Please enhance the user experience of `select all` in the modal, popup/warn the user that you select only the N first entries in case limit is reached, add tag button to automatically see ERC20, BEP20, QRC20, SmartChains or UTXO coins

If select all of a filtered research is less than `MaximumNbCoinsEnabled - portfolio_model.length` you can keep the current behavior, you can also show with hints, how much user can select coins

You can directly push to this pull request

Some example:

- I have 10 coins enabled, MaximumNbCoinsEnabled is equal to 50, i'm able to enable 40 (limit - nb_enabled_coins) assets

Please also adapt the code from best orders that propose to the user to enable a coin, same check should also occured here, add a well explained message in case user want to enable a coin but limit is reached.

@smk762 if you can provide a good english message for a new FAQ entry regarding simultaneous coins activated limitation (bandwith, performance, enable coins that you are willing to use) - it's will be fantastic

cc @tonymorony @smk762

### CODE REVIEW

**Comment by SylEze** [comment]
Steps:

- [ ] Visual settings to set the maximum number of enablable coins: [10, 20, 50, 75, 100, 150, 200]
- [ ] In enable coins modal, warn the user if he selects only the N first entries in case limit is reached
- [ ] Ensure `Send` button that requires a parent chain to be enabled, shows another modal than the enable one in case the limit is already reached. (e.g. https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/893)

File: atomic_defi_design/qml/Settings/SettingModal.qml [hunk 1]
```diff
@@ -156,6 +156,21 @@ Qaterial.Dialog {
...
+                                width: parent.width-30
+                                anchors.horizontalCenter: parent.horizontalCenter
+                                height: 50
+                                DexLabel {
+                                    Layout.alignment: Qt.AlignVCenter
+                                    Layout.fillWidth: true
+                                    text: qsTr("Maximum number of enabled coins")
+                                }
+                                DexComboBox {
+                                    model: [10, 20, 50, 75, 100, 150, 200]
```
Comment (Milerius) [inline atomic_defi_design/qml/Settings/SettingModal.qml:169]: Here it's should be possible to detect the maximum value, which is the global cfg size i believe

**Comment by SylEze** [comment]
> @SylEze every time i restart the current selection of combobox is empty
> image

Should be fixed in 5b039130760d9808e8ee9a0ac882eb34017ea151

**Comment by smk762** [comment]
Limit appears to be enforced as expected in standard usage, but if changing assets limit to a lower value, previously selected items remain selected, and attempting to activate from this state results in nothing being activated.

https://user-images.githubusercontent.com/35845239/117929918-f0d11080-b32f-11eb-8462-373d90acd113.mp4

Also suggest that the "Increase Assets Limit" button be renamed to "Change Assets Limit"

**Comment by Milerius** [comment]
> Limit appears to be enforced as expected in standard usage, but if changing assets limit to a lower value, previously selected items remain selected, and attempting to activate from this state results in nothing being activated.
> 
>  enable_limit.mp4 
> Also suggest that the "Increase Assets Limit" button be renamed to "Change Assets Limit"

Thanks you for the report ! yeah i agree in general @SylEze we can probably unselect all in case the limit is changed on the fly ^^ !

**Comment by Milerius** [comment]
> Limit appears to be enforced as expected in standard usage, but if changing assets limit to a lower value, previously selected items remain selected, and attempting to activate from this state results in nothing being activated.
> 
>  enable_limit.mp4 
> Also suggest that the "Increase Assets Limit" button be renamed to "Change Assets Limit"

Do you confirm that trying to enable coins from the send modal and best orders handle also the limit ?

**Comment by SylEze** [comment]
> Limit appears to be enforced as expected in standard usage, but if changing assets limit to a lower value, previously selected items remain selected, and attempting to activate from this state results in nothing being activated.
> enable_limit.mp4
> 
> Also suggest that the "Increase Assets Limit" button be renamed to "Change Assets Limit"

Thanks for your report, I will patch it asap

**Comment by SylEze** [comment]
> > Limit appears to be enforced as expected in standard usage, but if changing assets limit to a lower value, previously selected items remain selected, and attempting to activate from this state results in nothing being activated.
> > enable_limit.mp4
> > Also suggest that the "Increase Assets Limit" button be renamed to "Change Assets Limit"
> 
> Thanks for your report, I will patch it asap

Patched by 43157b05a610885e3c7dc777ed4b4ecb882c68c7

File: atomic_defi_design/qml/Wallet/EnableCoinModal.qml [hunk 2]
```diff
@@ -125,8 +130,32 @@ BasicModal {
...
+                        .arg(setting_modal.enableable_coins_count - API.app.portfolio_pg.portfolio_mdl.length)
+                        .arg(coin_cfg_model.checked_nb)
+            }
+        }
+
         // Buttons
         footer: [
+            DefaultButton {
+                property var enableable_coins_count: setting_modal.enableable_coins_count;
+                text: qsTr("Increase assets limit")
```
Comment (smk762) [inline atomic_defi_design/qml/Wallet/EnableCoinModal.qml:154]: Change this to "Change assets limit", as user might be decreasing it.

**Review by smk762 — APPROVED** [review]

File: atomic_defi_design/qml/Wallet/EnableCoinModal.qml (hunk 2 above)
Comment (Milerius) [reply atomic_defi_design/qml/Wallet/EnableCoinModal.qml:154]: implemented in https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/886/commits/2d32dcb85d16f8cf5ea3aece86a8710f49764646

**Pull request merged**